internal/db/db.go

```diff
@@ -302,6 +308,13 @@ func (db *db) DeleteDocActorRelationship(
 	return client.DeleteDocActorRelationshipResult{RecordFound: recordFound}, nil
 }
 
+func (db *db) GetNodeIdentity(context.Context) (immutable.Option[identity.PublicRawIdentity], error) {
```

Comment:
we won't return anything but public identity. I don't see a point of making the method name unnecessarily longer. And I don't understand why it's 'Todo'

Reply:
This is because to me `Identity` makes up of multiple components which includes the public,priv, etc. You also have a `PublicRawIndentity` as return type. I think it's quite fair to indicate it's just the public part of the identity, not the full identity.

Reply:
> You also have a PublicRawIndentity as return type.

This is exactly the reason why I don't see a point in stating it twice.

With this logic we should also change CLI to `defradb client public_node_identity`

Reply:
if for example a command like `defradb client node-identity get`

I would definitely prefer it to print:

```json
{
  "PublicIdentity": "XYZ"
}
```

Rather than:

```json
{
  "Identity": "XYZ"
}
```

Note: PrivateKey contains all the components of an identity to me as you can derive public key and `did` from it. So only saying `Identity` can be confusing if all it returns is the public part of the set identity.

However I am happy to concede this point if consensus disagrees with this nitpick of mine. Might be worth seeing what the others think as well.

Reply:
it prints:
```json
{
    "PublicKey": "XYZ",
    "DID": "did"
}
```

Reply:
Just noting that I prefer the current than the suggestion.  IMO there is no such thing as a public/private identity - only public and private components of the same identity, and as such I agree with the function name `GetNodeIdentity`, and the current print structure (essentially printing the public properties of the identity).

Reply:
> it prints:
> 
> ```json
> {
>     "PublicKey": "XYZ",
>     "DID": "did"
> }
> ```

Ah okay cheers! Will then just call this a `nitpick` and Andy agrees with the way it is so I concede this `nitpick`.